### PR TITLE
New ShapeBufferGeometry

### PR DESCRIPTION
**Pull request by Mugen87**
This PR introduces `ShapeBufferGeometry`. 

The first [DEMO](https://jsfiddle.net/op8nvLo4/2/) shows a simple shape geometry with a hole.
The second [DEMO](https://jsfiddle.net/jm2hwr19/2/) shows a geometry constructed out of two shapes.

Unfortunately i had to perform some changes in order to make `ShapeBufferGeometry` and `ShapeGeometry` more consistent to the new geometry entities.
- The public methods `addShape` and `addShapeList` were removed. The  correct way for creating a `ShapeBufferGeometry` is this:  1. Create your shape list.  2. Call the respective constructor. 3. If your shape list changes, recreate the geometry.
- `ShapeBufferGeometry` has no `options` parameter anymore. All parameters must be assigned explicitly.
- `ShapeBufferGeometry` dropped the support for the `material` parameter. If you pass an array of shapes, each shape automatically gets its own material index. You can check out this behavior in the second demo.
- There is no support for the `uvGenerator` parameter anymore. `ShapeBufferGeometry` creates "world uvs" by default. If users want to change this, they should maybe apply something like this:

``` javascript
UVWorldGenerator.generate( bufferGeometry );
```

I hope the implementation is okay so far. There was no need to change `webgl_geometry_shapes` to keep things working. What do you think?

### CODE REVIEW

File: src/geometries/ShapeBufferGeometry.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+	// build geometry
+
+	this.setIndex( ( indices.length > 65535 ? Uint32Attribute : Uint16Attribute )( indices, 1 ) );
+	this.addAttribute( 'position', Float32Attribute( vertices, 3 ) );
+	this.addAttribute( 'uv', Float32Attribute( uvs, 2 ) );
+
+	// generate normals
+
+	this.computeVertexNormals();
```
Comment (WestLangley) [inline src/geometries/ShapeBufferGeometry.js]: I would avoid calling `computeVertexNormals()` when you can compute the normals analytically. Are not all the normals `( 0, 0, 1 )`?

**Comment by WestLangley** [comment]
I would note in the example that you are compensating for the so-called "world UVs" by setting `offset/repeat` on the texture.

File: src/geometries/ShapeBufferGeometry.js (hunk 1 above)
Comment (Mugen87) [inline src/geometries/ShapeBufferGeometry.js]: You're right! It's actually better to create the normal values like the other buffers. The value is always `( 0, 0, 1 )`.

**Comment by Mugen87** [comment]
Done!

**Comment by mrdoob** [comment]
Many thanks!

**Pull request merged**